packages/verified-fetch/src/plugins/plugin-handle-dag-walk.ts

```diff
@@ -41,6 +42,10 @@ export class DagWalkPlugin extends BasePlugin {
         // send the 404 response
         return pathDetails
       }
+      if (options?.signal?.aborted === true) {
```

Comment:
Because you had to add this check at this high level, it means the execution context was suspended by the `await` of a promise while the signal fired it's abort event.

There's only one use of `await` in this method so that's where it has to be.